applications/crossbar/src/modules/cb_modules_util.erl

```diff
@@ -579,15 +579,40 @@ remove_plaintext_password(Context) ->
            ),
     cb_context:set_doc(Context, Doc).
 
--type assignment_updates() :: [{ne_binary(), knm_number:knm_number_return()}].
+-type assignment_update() :: {ne_binary(), knm_number:knm_number_return()}.
+-type assignment_updates() :: [assignment_update()].
 
 -spec apply_assignment_updates([{ne_binary(), api_binary()}]) ->
                                       assignment_updates().
 apply_assignment_updates(Updates) ->
-    [{DID, knm_number:assign_to_app(DID, Assign)}
+    [maybe_assign_to_port_number(DID, Assign)
      || {DID, Assign} <- Updates
     ].
 
+-spec maybe_assign_to_port_number(ne_binary(), api_binary()) ->
+                                         assignment_update().
+maybe_assign_to_port_number(DID, Assign) ->
+    Num = knm_converters:normalize(DID),
+    case knm_port_request:get(Num) of
+        {'error', _} ->
+            {DID, knm_number:assign_to_app(DID, Assign)};
+        {'ok', JObj} ->
+            NewPortJObj = update_port_request(Num, Assign, JObj),
+            {DID, kz_datamgr:save_doc(?KZ_PORT_REQUESTS_DB, NewPortJObj)}
```

Comment:
Just noticed: this is `{ne_binary(), {ok, kz_json:object()} | {error, any()}}`, which does not map to `assignment_update()`.
Also this assignment code should probably go to knm_number.erl as there already are ports stuff there, and this has to do with general use of ports.